crates/sui-source-validation/src/lib.rs

```diff
@@ -141,11 +167,13 @@ impl<'a> BytecodeSourceVerifier<'a> {
         compiled_package: &CompiledPackage,
         verify_deps: bool,
         source_mode: SourceMode,
-    ) -> Result<(), SourceVerificationError> {
+    ) -> Result<(), AggregateSourceVerificationError> {
         // On-chain address for matching root package cannot be zero
         if let SourceMode::VerifyAt(root_address) = &source_mode {
             if *root_address == AccountAddress::ZERO {
-                return Err(SourceVerificationError::ZeroOnChainAddresSpecifiedFailure);
+                return Err(AggregateSourceVerificationError(vec![
+                    SourceVerificationError::ZeroOnChainAddresSpecifiedFailure,
+                ]));
```

Comment:
You could take advantage of your `From` implementation here:

```suggestion
                return Err(SourceVerificationError::ZeroOnChainAddresSpecifiedFailure.into());
```

Reply:
![](https://media.tenor.com/ySPZZZr5Vm0AAAAM/amazed-face.gif)